extensions/wikia/PageHeader/Subtitle.class.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+class Subtitle {
+	/**
+	 * @var string
+	 */
+	public $subtitle;
+
+	/**
+	 * @var string
+	 */
+	public $pageSubtitle;
+
+	/**
+	 * @var mixed|null
+	 */
+	public $supressPageSubtitle;
+
+	public function __construct( \WikiaApp $app ) {
+		$this->supressPageSubtitle = $app->wg->SupressPageSubtitle;
+		if ( !$this->supressPageSubtitle ) {
+			$this->subtitle = $this->getSubtitle();
+			//watch list uses that, pageSubject?
+			$this->pageSubtitle = $this->handlePageSubtitle();
+		}
+	}
+
+	private function getSubtitle() {
+		$wgOutput = \RequestContext::getMain()->getOutput();
+
+		return $wgOutput->getSubtitle();
+	}
+
+	private function handleTalkPage() {
+		$title = \RequestContext::getMain()->getTitle();
+
+		if ( $title->isTalkPage() ) {
+			$namespace = $title->getNamespace();
+
+			// back to subject article link
+			switch ( $namespace ) {
+				case NS_TEMPLATE_TALK:
+					$msgKey = 'oasis-page-header-back-to-template';
+					break;
+
+				case NS_MEDIAWIKI_TALK:
+					$msgKey = 'oasis-page-header-back-to-mediawiki';
+					break;
+
+				case NS_CATEGORY_TALK:
+					$msgKey = 'oasis-page-header-back-to-category';
+					break;
+
+				case NS_FILE_TALK:
+					$msgKey = 'oasis-page-header-back-to-file';
+					break;
+
+				default:
+					$msgKey = 'oasis-page-header-back-to-article';
+			}
+
+			return \Linker::link(
+				$title->getSubjectPage(),
+				wfMessage( $msgKey )->escaped(),
+				[ 'accesskey' => 'c' ]
+			);
+		}
+
+		return null;
+	}
+
+	private function getPageType() {
+		$title = \RequestContext::getMain()->getTitle();
+		$namespace = $title->getNamespace();
+
+		$pageType = null;
+		// render page type info
+		switch ( $namespace ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "case default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)